src/ipred.rs

```diff
@@ -1200,7 +1200,8 @@ unsafe fn ipred_filter_rust<BD: BitDepth>(
     for y in (0..height).step_by(2) {
         let topleft_off = topleft_off - y;
         let mut topleft = topleft_in[topleft_off];
-        for x in (0..width).step_by(4) {
+        for x in 0..width / 4 {
+            let x = x * 4;
```

Comment:
Hmm, I might be totally wrong.  I forgot exactly what I was trying to optimize.

Reply:
Oh yeah, I think I wanted to make sure `width` is divisible by 4, so that `top[x..][..4]` where `top.len() == width` doesn't need a check.  If `width % 4 != 0`, then `x` could be `width - 1`, but if `width % 4 == 0`, then it guarantees that the max `x` is `width - 4`.  Let me fix this.

Reply:
Fixed in [`cdc3c5e..59ab627`](https://github.com/memorysafety/rav1d/compare/cdc3c5e32b4e84bd0da00dca204bea2b740d1a60..59ab62729478ac84d58b19b8460ed41612dc3694).